packages/liquidation-sdk-viem/src/preLiquidation/positionGetters.ts

```diff
@@ -4,69 +4,99 @@ import { fetchAccrualPosition } from "@morpho-org/blue-sdk-viem";
+    const data = (await response.json()) as {
+      results: PreLiquidationData[];
+    };
 
-      if (
-        loanAsset === undefined ||
-        collateralAsset === undefined ||
-        collateralAsset === null
-      )
-        return;
+    const preLiquidationInstances = await Promise.all(
+      data.results
+        .filter((preLiquidation) => preLiquidation.price !== null)
+        .map((preLiquidation) => {
+          return {
+            ...preLiquidation,
+            price: BigInt(preLiquidation.price!),
+            preLiquidationParams: {
+              ...preLiquidation.preLiquidationParams,
+              preLCF1: BigInt(preLiquidation.preLiquidationParams.preLCF1),
+              preLCF2: BigInt(preLiquidation.preLiquidationParams.preLCF2),
+              preLIF1: BigInt(preLiquidation.preLiquidationParams.preLIF1),
+              preLIF2: BigInt(preLiquidation.preLiquidationParams.preLIF2),
+              preLltv: BigInt(preLiquidation.preLiquidationParams.preLltv),
+            },
+          };
+        })
+        .map(async (preLiquidation) => {
+          const {
+            markets: { items: market },
+          } = await apiSdk.getMarketAssets({
+            chainId,
+            marketId: preLiquidation.marketId,
+          });
 
-      return {
-        preLiquidation,
-        borrowers: await authorizationLogs(client, preLiquidation),
-        loanAsset,
-        collateralAsset,
-      };
-    }),
-  );
+          const loanAsset = market !== null ? market[0]?.loanAsset : undefined;
+          const collateralAsset =
+            market !== null ? market[0]?.collateralAsset : undefined;
 
-  const preLiquidablePositions = await Promise.all(
-    preLiquidationInstances
-      .filter((position) => position !== undefined)
-      .map(async (preLiquidationPosition) => {
-        return await Promise.all(
-          preLiquidationPosition.borrowers.map(async (borrower) => {
-            return await getPreLiquidablePosition(
-              client,
-              preLiquidationPosition.preLiquidation,
-              borrower,
-              preLiquidationPosition.collateralAsset,
-              preLiquidationPosition.loanAsset,
-            );
-          }),
-        );
-      }),
-  );
+          if (
+            loanAsset === undefined ||
+            collateralAsset === undefined ||
+            collateralAsset === null
+          )
+            return;
+
+          return {
+            ...preLiquidation,
+            loanAsset,
+            collateralAsset,
+          };
+        }),
+    );
+
+    const preLiquidablePositions = await Promise.all(
+      preLiquidationInstances
+        .filter((position) => position !== undefined)
+        .map(async (preLiquidationPosition) => {
+          return await Promise.all(
+            preLiquidationPosition.enabledPositions.map(async (borrower) => {
+              return await getPreLiquidablePosition(
```

Comment:
```suggestion
        .map((preLiquidationPosition) => Promise.all(
            preLiquidationPosition.enabledPositions.map((borrower) => getPreLiquidablePosition(
```

No need to async/await when you return a promise (it's overwhelming to review), but i'm not that opinionated on this)